expression/builtin_string.go

```diff
@@ -1589,9 +1589,81 @@ type builtinFormatSig struct {
 	baseBuiltinFunc
 }
 
+type formatFunc func(float64, int64) string
+
 // See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_format
 func (b *builtinFormatSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("format")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	sc := b.ctx.GetSessionVars().StmtCtx
+	arg0, err := args[0].ToDecimal(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	arg1, err := args[1].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	var arg2 string
+
+	// TODO: Neet to be moved away.
+	supportedFormat := map[string]formatFunc{
+		"en-US": _formatENUS,
+		"zh-CN": _formatZHCN,
+	}
+	if len(args) == 3 {
+		arg2, err = args[2].ToString()
+	}
+	_, exist := supportedFormat[arg2]
+	if !exist {
+		arg2 = "en-US"
+	}
+	number, err := arg0.ToFloat64()
```

Comment:
Will this operation loss precision? `SELECT FORMAT(12332.1234567890123456789012345678901, 22);`
PTAL @coocood